src/internal/validation/argument_validator.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Argument Validation
+""" 
+
+from enums.worker import WorkerType, WorkerStatus
+from exceptions.invalid_parameter import InvalidParamException
+
+
+class ArgumentValidator(object):
+    """
+    Helper method for validating an argument that will be used by this API in any requests.
+    """
+    __instance = None
+    
+    @staticmethod 
+    def getInstance():
+      """ Static access method. """
+      if ArgumentValidator.__instance == None:
+         ArgumentValidator()
+      return ArgumentValidator.__instance
+    
+    
+    def __init__(self):
+      """ Virtually private constructor. """
+      self.enum_dic = {
+            "WorkerType" : WorkerType,
+            "WorkerStatus" : WorkerStatus
+        }
+      if ArgumentValidator.__instance != None:
+         raise Exception("This class is a singleton!")
+      else:
+        ArgumentValidator.__instance = self
+
+
+    def enum_value(self,id, enum, value):
```

Comment:
function, parameter, return value description missing in this and other functions in this file.

Reply:
done